Assets/com.fluid.behavior-tree/Runtime/TaskParents/Composites/Parallel.cs

```diff
@@ -1,52 +1,25 @@
-﻿using System.Collections.Generic;
 using CleverCrow.Fluid.BTs.Tasks;
 
 namespace CleverCrow.Fluid.BTs.TaskParents.Composites {
     public class Parallel : CompositeBase {
-        private readonly Dictionary<ITask, TaskStatus> _childStatus = new Dictionary<ITask, TaskStatus>();
-
         public override string IconPath { get; } = $"{PACKAGE_ROOT}/CompareArrows.png";
 
         protected override TaskStatus OnUpdate () {
-            var successCount = 0;
-            var failureCount = 0;
-
-            foreach (var child in Children) {
-                TaskStatus prevStatus;
-                if (_childStatus.TryGetValue(child, out prevStatus) && prevStatus == TaskStatus.Success) {
-                    successCount++;
-                    continue;
-                }
-
-                var status = child.Update();
-                _childStatus[child] = status;
-
-                switch (status) {
-                    case TaskStatus.Failure:
-                        failureCount++;
-                        break;
-                    case TaskStatus.Success:
-                        successCount++;
-                        break;
-                }
-            }
-
-            if (successCount == Children.Count) {
-                End();
-                return TaskStatus.Success;
-            }
-
-            if (failureCount > 0) {
+            TaskStatus status = TaskStatus.Success;
+            foreach(var child in Children)
+                status |= child.Update();
```

Comment:
This is a great idea. But it will retick a successful node, which is incorrect behavior. We should instead probably have a list that pops off successful nodes and removes them. Note the test suite has a test for this and it probably fails right now.

Reply:
So we don't have to modify `TaskStatus` can always have a flag set here as so (rough pseudo code below). It's a few more lines but I want to keep the statuses as vanilla enums. There definitely need to be tests with this to verify integrity of the tick process.

```
var status = Success;

// @NOTE Child stack needs to be cleared on Reset and re-populated on start (see other composite setups)
for (child in childStack) {
  var childStatus = child.Update();

  // Failure detected, we don't care about anything
  if (status == failure) continue;

  // The child was successful, remove it
  if (childStatus == success) {
    childStack.remove(child);
  } else {
    // Could be continue or failure, record it as an override of the success status
    status = childStatus;
  }
}
```